iothub_client/tests/iothubclient_fuzz_amqp/iothubclient_fuzz.c

```diff
@@ -0,0 +1,685 @@
+            break;
+
+        case 3: // client performative Open
+            // server begin
+            //<- [OPEN]* {DeviceGateway_5ad507ef78724b7386c9c3309fb31014,localhost,65536,8191,240000,NULL,NULL,NULL,NULL,NULL}
+            size_read = load_from_file(4, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 4: // send begin
+            //<- [BEGIN]* {0,1,5000,4294967295,262143,NULL,NULL,NULL}
+            size_read = load_from_file(5, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 5: // send attach $cbs-sender
+            //<- [ATTACH]* {$cbs-sender,0,true,0,0,* {$cbs,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL},* {$cbs,NULL,NULL,NULL,NULL,NULL,NULL},NULL,NULL,NULL,1048576,NULL,NULL,NULL}
+            size_read = load_from_file(6, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 6: // send attach $cbs-receiver
+            //<- [ATTACH]* {$cbs-receiver,1,false,0,0,* {$cbs,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL},* 
+            size_read = load_from_file(8, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            //< -[FLOW] * {0, 5000, 1, 4294967295, 0, 0, 100, 0, NULL, false, NULL}
+            size_read = load_from_file(7, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 7: // send_flow
+            //<- [DISPOSITION]* {true,0,NULL,true,* {},NULL}
+            size_read = load_from_file(9, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 8: // session_send_transfer
+            // send SAS token to hub
+            //<- [TRANSFER]* {1,0,<01 00 00 00>,0,NULL,false,NULL,NULL,NULL,NULL,false} 
+            size_read = load_from_file(10, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 9: // send_disposition
+            break;
+
+        case 10: // send_attach (events)
+            //<- [ATTACH]* {link-snd-eeqq-6322c15e-2c82-4a42-a57d-aa203cf1f4dc,2,true,0,NULL,* {link-snd-eeqq-6322c15e-2c82-4a42-a57d-aa203cf1f4dc-source,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL},* {amqps://ericwol-hub.azure-devices.net/devices/eeqq/messages/events,NULL,NULL,NULL,NULL,NULL,NULL},NULL,NULL,NULL,1048576,NULL,NULL,{[com.microsoft:client-version:iothubclient/1.3.9 (native; WindowsProduct:0x00000004 6.2; x64; {F9FA04EF-2602-43AB-8505-A1EDE028ADD8})]}}
+            size_read = load_from_file(11, file_buffer, sizeof(file_buffer));
+            if (size_read > 0x5f + 41 && size > 31 + 41)
+            {
+                memcpy(&file_buffer[0x1f], &((char*)buffer)[31], 41);
```

Comment:
curious what all these hardcoded hex numbers represent?

Reply:
it's the GUID part of the link source link-snd-eeqq-**6322c15e-2c82-4a42-a57d-aa203cf1f4dc**-source